src/brod_consumer.erl

```diff
@@ -442,17 +476,50 @@ handle_reset_offset(#state{offset_reset_policy = Policy} = State, _Error) ->
                   reset_to_latest   -> ?OFFSET_LATEST
                 end,
   State1 = State#state{ begin_offset = BeginOffset
-                      , pending_acks = []
+                      , pending_acks = #pending_acks{}
                       },
   {ok, State2} = resolve_begin_offset(State1),
   NewState = maybe_send_fetch_request(State2),
   {noreply, NewState}.
 
-handle_ack([], _Offset) -> [];
-handle_ack([H | Offsets], Offset) ->
-  case H =< Offset of
-    true  -> handle_ack(Offsets, Offset);
-    false -> [H | Offsets]
+handle_ack(#pending_acks{ offsets_queue = ?EMPTY_QUEUE
+                        , queue_rear = {Begin, End}
+                        , count = Count} = PendingAcks, Offset) ->
+  if
+    Offset >= End ->
+      NewCount = Count - (End - Begin + 1),
```

Comment:
because it's inclusive at both ends? i.e. it's a range of `[Begin, End]` not `[Begin, End)`